pkg/runtime/adoption_reconciler_test.go

```diff
@@ -435,9 +437,13 @@ func assertAWSResourceCreation(
 ) {
 	if expectedCreation {
 		kc.AssertCalled(t, "Create", ctx, res.RuntimeObject())
+		res.AssertCalled(t, "DeepCopy")
+		res.AssertCalled(t, "SetStatus", res)
 		statusWriter.AssertCalled(t, "Update", ctx, res.RuntimeObject())
 	} else {
 		kc.AssertNotCalled(t, "Create", ctx, res.RuntimeObject())
+		res.AssertNotCalled(t, "DeepCopy")
+		res.AssertNotCalled(t, "SetStatus", res)
```

Comment:
Not to be picky, but I don't think these tests really *test* the functionality. It's the order in which these are called, and how we manipulate the underlying pointers, that is essential to this PR doing what it intends to do. Simply seeing it call `DeepCopy` does not validate that it's successful.

Reply:
Fair point, I have updated the test to at least use a different object for DeepCopy.
The new assertions test that SetStatus is invoked after retrieving the DeepCopy object.